app/src/main/java/com/halim/adam/cashmaster/DatabaseHelper.java

```diff
@@ -0,0 +1,88 @@
+package com.halim.adam.cashmaster;
+
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.os.Debug;
+import android.util.Log;
+
+import java.util.Date;
+
+public class DatabaseHelper extends SQLiteOpenHelper {
+    private static final String DATABASE_NAME = "cash_master.db";
+    private static final int DATABASE_VERSION = 3;
+
+    public DatabaseHelper(Context context){
+        super(context, DATABASE_NAME,null, DATABASE_VERSION);
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        String sql = "CREATE TABLE category ( id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT UNIQUE );";
+        db.execSQL(sql);
+        sql = "CREATE TABLE income ( id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, price NUMERIC, date TEXT DEFAULT (date('now')));";
+        db.execSQL(sql);
+        sql = "CREATE TABLE jar ( id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, portion INTEGER );";
+        db.execSQL(sql);
+        sql = "CREATE TABLE spending ( id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, date TEXT DEFAULT (date('now')), price NUMERIC, categoryId INTEGER, jarId INTEGER, FOREIGN KEY(categoryId) REFERENCES category(id) ON DELETE SET NULL );";
+        db.execSQL(sql);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
+
+    }
+
+    /*
+    INSERT TO TABLES ---------------------------------------------------------------------------------------------
+     */
+
+    /**
+     * Insert to table 'category'
+     * @param name name of the category
+     */
+    public void InsertCategory(String name){
+        SQLiteDatabase db = this.getWritableDatabase();
+        String sql = "INSERT INTO category (name) VALUES ('" + name + "');";
```

Comment:
You might not want to add user input directly to SQL statement like this. A better approach is to use parameterized statements (SQLite Statements) like this

https://stackoverflow.com/a/29797229